### PR TITLE
chore: remove replicache-nextjs folder and replace with library

### PR DESCRIPTION
**Pull request by cesara**

### CODE REVIEW

File: pages/d/[id].tsx [hunk 1]
```diff
@@ -2,8 +2,8 @@ import { GetServerSideProps } from "next";
 
 import App from "../../src/app";
 
-import { spaceExists } from "../../replicache-nextjs/api";
-import { useReplicache } from "../../replicache-nextjs/frontend/use-replicache";
+import { spaceExists } from "replicache-nextjs/out/backend";
```
Comment (aboodman) [inline pages/d/[id].tsx]: Instead of having the export be `out/backend` can we have it be just `backend`?

File: package.json [hunk 2]
```diff
@@ -29,7 +28,6 @@
     "classnames": "^2.3.1",
     "mocha": "^9.2.1",
     "nanoid": "^3.3.1",
-    "pg-mem": "^2.5.0",
```
Comment (aboodman) [inline package.json:32]: Can the supabase-js dep go away too?

**Review by aboodman — COMMENTED** [review]
Very cool

File: pages/d/[id].tsx (hunk 1 above)
Comment (cesara) [reply pages/d/[id].tsx]: This is a TS issue that ignored exports in package.json, I believe is fixed in 4.7. I'll try it out.

File: package.json (hunk 2 above)
Comment (cesara) [reply package.json:32]: yup i'll remove

File: pages/d/[id].tsx (hunk 1 above)
Comment (cesara) [reply pages/d/[id].tsx]: I tried my best to fix this issue, but was not successful there is a 200comment thread about this here 
https://github.com/microsoft/TypeScript/issues/33079. @grgbkr recommended I just rename the out to lib instead and we can figure it out later.

**Pull request merged**